src/views/blog-post/series/series-toc.astro

```diff
@@ -14,93 +15,99 @@ interface SeriesTocProps {
 
-const isActiveInMiddle = middlePosts.some(
-	(middlePost) => middlePost.order === post.order
+const showMoreText = translate(
+	Astro,
+	"action.view_all_chapters",
+	"" + activePostsMeta.length
 );
 ---
 
 <div class={styles.seriesTableOfContent}>
 	<div class={styles.seriesHeader}>
 		{
-			collection ? (
-				<a href={`/collections/${collection.slug}`}>
-					Part of our series: {collection.title}
-				</a>
-			) : (
-				`Part of our series: ${post.collection}`
+			(
+				<h2 class={styles.titleContainer} id="series-header">
+					<span class={`text-style-body-medium-bold ${styles.partOfSeries}`}>
+						This article is part of a series:
+					</span>
+					{collection ? (
+						<a
+							class={`text-style-headline-4 ${styles.articleTitle}`}
+							href={`/collections/${collection.slug}`}
+						>
+							{collection.title}
+						</a>
+					) : (
+						<span class={`text-style-headline-4 ${styles.articleTitle}`}>
+							{post.collection}
+						</span>
+					)}
+				</h2>
 			)
 		}
 	</div>
-	<ol aria-labelledby="series-header" role="list">
-		{
-			firstPosts.map((seriesPost) => {
-				const isActive = post.order === seriesPost.order;
-
-				return (
-					<SeriesToCListItem
-						key={seriesPost.slug}
-						isActive={isActive}
-						post={seriesPost}
-					/>
-				);
-			})
-		}
-		{
-			middlePosts.length !== 0 ? (
-				<>
-					<li role="listitem" class={isActiveInMiddle ? styles.isActive : ""}>
-						<button aria-expanded={false} data-in-middle-toggle="true">
-							{middlePosts.length} more parts
-						</button>
-					</li>
-					{middlePosts.map((seriesPost) => {
-						const isActive = post.order === seriesPost.order;
-
-						return (
-							<SeriesToCListItem
-								key={seriesPost.slug}
-								isActive={isActive}
-								post={seriesPost}
-								inMiddle={true}
-							/>
-						);
-					})}
-				</>
-			) : null
-		}
-		{
-			lastPosts.map((seriesPost) => {
-				const isActive = post.order === seriesPost.order;
-
-				return (
-					<SeriesToCListItem
-						key={seriesPost.slug}
-						isActive={isActive}
-						post={seriesPost}
-					/>
-				);
-			})
-		}
-	</ol>
+	<div class={styles.contentsContainer}>
+		<ol
+			aria-labelledby="series-header"
+			role="list"
+			class={styles.listContainer}
+		>
+			{
+				activePostsMeta.length !== 0 ? (
+					<>
+						{activePostsMeta.map((seriesPost) => (
+							<li
+								class={styles.navigationItemOuter}
+								data-dont-show-initially={!seriesPost.shouldShowInitially}
+							>
+								<a
+									href={`/posts/${seriesPost.slug}`}
+									class={`text-style-body-large-bold ${styles.navigationItem}`}
+									data-is-active={seriesPost.isActive}
+								>
+									Part {seriesPost.order}: {getShortTitle(seriesPost)}
+								</a>
+							</li>
+						))}
+					</>
+				) : null
+			}
+		</ol>
+		<div class={styles.buttonContainer}>
+			<Button id="toggleAllButton" tag="button">
+				{showMoreText}
+			</Button>
+		</div>
```

Comment:
This button should likely be optional for collections with less than 4 posts, since all posts in the collection would be visible regardless
(can be seen on https://unicorn-utterances-git-uwu-series-toc-oceanbit.vercel.app/posts/minecraft-data-packs-introduction)

Reply:
Oh, uh, right, forgot. 😅